pkg/controllers/managedzone/managedzone_controller.go

```diff
@@ -164,7 +163,11 @@ func (r *ManagedZoneReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 func (r *ManagedZoneReconciler) publishManagedZone(ctx context.Context, managedZone *v1alpha1.ManagedZone) error {
 
-	mzResp, err := r.DNSProvider.EnsureManagedZone(managedZone)
+	DNSProvider, err := r.DNSProvider(ctx, managedZone)
+	if err != nil {
```

Comment:
I think we could check whether this is a not found err, and if it is update the status of the zone with an error condition

Reply:
Actually, this gets checked on line 107 so would it not cause an issue with both trying to update it?